src/main/java/com/blackduck/integration/detect/ApplicationUpdater.java

```diff
@@ -654,30 +668,114 @@ private File download(File installDirectory, HttpUrl downloadUrl, String current
+
         currentInstalledVersion = getVersionFromDetectFileName(currentInstalledVersion);
-        if (response.isStatusCodeSuccess() && newVersionString != null) {
-            logger.debug("{} Old version: {}, New Version: {}", LOG_PREFIX, currentInstalledVersion, newVersionString);
-            if (StringUtils.isNotBlank(newVersionString)
-                && !newVersionString.equals(currentInstalledVersion)
-                && !isDownloadVersionTooOld(currentInstalledVersion, newVersionString)) {
-                return handleSuccessResponse(response, installDirectory.getAbsolutePath(), newVersionString);
+        if (shouldUpdate(currentInstalledVersion, newVersionString)) {
+            if (potentialNewJar == null) {
+                potentialNewJar = downloadNewJar(response, installDirectory, newFileName); 
+                
+                if (potentialNewJar == null) {
+                    logger.warn("Failed to download the new Detect JAR.");
+                    return null;
+                }
             }
-        } else if (response.getStatusCode() == HttpStatus.SC_NOT_MODIFIED) {
-            logger.info("{} Present Detect installation is up to date - skipping download.", LOG_PREFIX);
+            
+            return validateDownloadedJar(potentialNewJar);
         } else {
-            String problemUrl = captureProblemDetectUrl(downloadUrl);
-            String message = StringUtils.isNotBlank(response.getStatusMessage()) ? response.getStatusMessage() : EnglishReasonPhraseCatalog.INSTANCE.getReason(response.getStatusCode(), Locale.ENGLISH);
-            logger.warn("{} Unable to download artifact from {}.", LOG_PREFIX, problemUrl);
-            logger.warn("{} Response code from {} was: {} {}", LOG_PREFIX, problemUrl, response.getStatusCode(), message);
+            logger.info("New version {} is not applicable for update. Using existing version {} instead.", newVersionString, currentInstalledVersion);
         }
         return null;
     }
+
+    public boolean isValidSemVer(String version) {
+        Pattern pattern = Pattern.compile(SEMVER_PATTERN);
+        Matcher matcher = pattern.matcher(version);
+        return matcher.matches();
+    }
     
+    private File downloadNewJarIfNeeded(Response response, File installDirectory, String newFileName, String newVersionString) throws IOException, IntegrationException {
+        if (newVersionString == null) {
+            return downloadNewJar(response, installDirectory, newFileName);
+        }
+        return null;
+    }
+
+    private void handleFailure(Response response, HttpUrl downloadUrl) throws IOException {
+        String problemUrl = getTrueDetectDownloadUrl(downloadUrl);
+        String message = StringUtils.isNotBlank(response.getStatusMessage()) ? response.getStatusMessage() : EnglishReasonPhraseCatalog.INSTANCE.getReason(response.getStatusCode(), Locale.ENGLISH);
+        logger.warn("{} Unable to download artifact from {}.", LOG_PREFIX, problemUrl);
```

Comment:
Since we refer specifically to the artifact as a jar elsewhere, maybe do the same here?
"{} Unable to download jar from {}."